### DISCUSSION TITLE
Bug?: Rules defined in subsequent config object isn't overriding first config object (FlatConfig)

### DISCUSSION THREAD

**Discussion by michaelfaith** [eslint/eslint#17443] [Q&A]
I'm working on creating a config for my team, and a big goal of ours is to use the new Flat Config and ESM.  As a base config, I have the eslint recommended config with no file filter.  And then add to that a config object that narrows the focus to .ts flavors and adds recommended rules from the typescript plugin, which turns off some of the eslint recommended (e.g. `no-unused-vars`).  However, those rules are still firing, when run on ts files.  Any thoughts on why these aren't being turned off?  Based on [this section](https://eslint.org/docs/latest/use/configure/configuration-files-new#rule-configuration-cascade) in the docs, I would have expected the subsequent config to override the earlier config.

example config:
```js
import js from '@eslint/js';
import tsPlugin from '@typescript-eslint/eslint-plugin';
import tsParser from '@typescript-eslint/parser';

export default [
  js.configs.recommended,  // Applied to all (includes 'no-unused-vars')
  {
    files: ['**/*.{ts,mts,cts,tsx,mtsx}'],
    plugins: { '@typescript-eslint': tsPlugin },
    languageOptions: {
      parser: tsParser,
      parserOptions: { sourceType: 'module' },
    },
    rules: {
      ...tsPlugin.configs['eslint-recommended'].rules,
      ...tsPlugin.configs.recommended.rules,  // This turns off 'no-unused-vars'
    },
  }
];
```

I'm thinking this might be a bug, but wanted to make sure I'm not setting this up incorrectly.

**Comment by mdjermanovic** [comment]
Can you try this:

```diff
-files: ['*.ts', '*.mts', '*.cts', '*.tsx', '*.mtsx'],
+files: ['**/*.ts', '**/*.mts', '**/*.cts', '**/*.tsx', '**/*.mtsx'],
```

In the Flat Config, `'*.ts'` matches only at the top level (e.g., it matches `a.ts`, but not `src/a.ts`), so if you're testing on a file that isn't top-level then the config object isn't applied.

**Reply by michaelfaith** [reply]
Oops, that was an error on my part, transcribing the example.  We actually have this:

```js
files: ['**/*.{ts,mts,cts,tsx,mtsx}'],
```

So i think that would be correctly formatted?  I updated the op

**Comment by michaelfaith** [comment]
I was able to figure out my issue with this.  The error was on my side.  It's working as expected.  Apologies for the false alarm
